### DISCUSSION TITLE
Convection boundary conditions

### DISCUSSION THREAD

**Discussion by tusharve** [zwicker-group/py-pde#154] [Q&A]
I am trying to set up a convection boundary on 3 surfaces of a rectangle 2D mesh. 

What's the boundary condition syntax for that? I need conditions to be T(x,t)= hAdelT where delT is difference between surface temperature and ambient temperature.

Any help would be appreciated!

**Comment by tusharve** [comment]
> I am trying to set up a convection boundary on 3 surfaces of a rectangle 2D mesh. 
> 
> What's the boundary condition syntax for that? I need conditions to be T(x,t)= hAdelT where delT is difference between surface temperature and ambient temperature.
> 
> Any help would be appreciated!

For more explanation, I'm trying to set up a solid cylinder generating heat in contact with a  273K cold plate at the bottom and 296K temperature air on the other sides. Radial and axial conductivities are different

**Reply by david-zwicker** [reply]
Could please specify the exact mathematical formulation of your boundary condition. I don't know how to parse `T(x,t)= hAdelT`.

**Reply by tusharve** [reply]
Hi David, I'm setting up a convection boundary condition on 3 sides of a 2 D rectangular system and a cold plate conduction boundary on the 4th side.

So the conditions would be,

Q- heat flux
h- heat convection coefficient
delT- temperature difference between plate surface and ambient air
A-area of the plate (assuming unit thickness)

So the boundary condition should be,

![image](https://user-images.githubusercontent.com/29714668/138140297-b9613311-d664-4076-b698-2c5c54986f26.jpeg)


Q=h*A*delT

I have attached a copy of the boundary code I'm using currently. 

![image](https://user-images.githubusercontent.com/29714668/138139765-abc1e6cc-fb03-4e1b-88ae-c162b674d57e.jpeg)

I'm also trying to simulate the Matlab code [here](https://petermattia.com/articles/2017/11/09/thermal-modeling-cylindrical-batteries.html) but with an axial component with a different thermal conductivity.

I appreciate the help. Thanks!

**Reply by david-zwicker** [reply]
I think you are very close already. You indeed need a mixed boundary condition for this case. `py-pde` expects the BC, to be of the form `∂c/∂n + A * c = B` where the derivative is the normal derivative at the boundary. You thus need to bring you BC into this form. If you then have fixed coefficients `A` and `B`, you can simply use the boundary condition `{'type': 'mixed', 'value': A, 'const': B}`.

**Reply by tusharve** [reply]
Thanks David! I think this has solved my problem for the time being. I might develop a more detailed model in the near future.
